models/AppModel.js

```diff
@@ -0,0 +1,64 @@
+var AppModel = Backbone.Model.extend({
+  defaults: function () {
+    return {
+      videos: new VideosCollection(),
+      currentVideo: null,
+      currentSearchTerm: 'cats',
+      endOfResults: false
+    }
+  },
+
+  initialize: function () {
+    this.listenTo(this.get("videos"), "update", this.setCurrentVideo)
+    this.listenTo(this.get("videos"), "update", this.checkForEndofResults)
+    //When line below is uncommented, completes an initial search (of "cats") so that the page will not be empty when user loads the page
+    this.get("videos").searchVideos(this.get("currentSearchTerm"))
+  },
+
+  //Update the current search term and call on the videos collection to search for new videos with that search term
+  //The type parameter is needed since both normal searches for a collection of five or more related videos returned and a search for single video id utilize this method. The default type is "searchTerm" used for returning multiple results. An individual video id search will have a type of "videoId"
+  updateSearchTerm: function (searchTerm, type = "searchTerm") {
+    this.set("currentSearchTerm", searchTerm)
+    var numberOfResults = type === "searchTerm" ? 5 : 1;
+    this.get("videos").searchVideos(this.get("currentSearchTerm"), numberOfResults)
+  },
+
+  //Sets the current Video attribute. This method could either be called from a new search taking place or from the user clicking one of the videos on the side bar
+  setCurrentVideo: function (input) {
+    //If this method is called from the user clicking a video on the sidebar then input will be a string of the video's id. Otherwise if the method is called from a new search taking place then input will be the new videoCollections
+    var newCurrentVideo;
```

Comment:
Nit-picky, but this pattern of declaring a variable to be defined later is usually frowned upon.